### PR TITLE
Extend/rework security-related documentation

### LINKED ISSUE #3448: doc/lxd-ssl-authentication.md unclear
**This is a documentation ticket.** I noticed that the docs on "LXD REST API's security done right" is a bit sparse, spread over multiple files and still requiring to delve deep to understand its possibilities and caveats. Assuming a server with lxd called **mightyserv** and a devs notebook with lxd called **devnb** and a setup along the following:

```
### setup mightyserv
mightyserv # lxc config set core.https_address 192.168.1.100:48443 # listen on a nonstandard port
mightyserv # lxc config set core.trust_password MYSECRETPASS
### review mightyserv
mightyserv # lxc config get core.https_address
mightyserv # lxc config get core.trust_password
# setup devnb
lxc remote add mightyserv 192.168.1.100:48443
### review devnb
devnb # lxc remote list
devnb # lxc remote list mightyserv:
devnb # lxc list mightyserv:
```

i found it hard(er) to find info on the following:

1. when are certs generated in ~/.config/lxc (on devdb)?
2. how are these certs generated, how can one force their regeneration (i.e. if the used cipher would develop a major vulnerability)
3. where are client certificates stored (on mightyserv)?
4. how do you disable client's (devnb) to mightyserve, when devdb is identified with its specific certificate? **lxc trust remove lxc config trust list**
5. how do you list client certificates? **lxc config trust list**
6. why would one not want to open the api port to the world if access to it is protected with a strong cipher (https://github.com/lxc/lxd/blob/master/doc/lxd-ssl-authentication.md)? why bother with a ssh tunnel (https://stgraber.org/2016/04/12/lxd-2-0-remote-hosts-and-container-migration-612/#comment-235914) if access to the api should be similarly secure as with key-based ssh access to the server? (apart from the obvious reasons such as doubling the security layers and protecting from dos attacks)
7. how does the api server prevent dos attacks? are there i.e. some fail2ban rules that can be used, or is there a way to white-list access from certain ips?
8. what do people actually use in production to secure their lxd-api instances?
9. how do you disable adding new clients (on mightyserv)
10. best practices, or what does the developer team use in production (everything from nonstadanard api port to guarding servers with war elephants)

Maybe additions to https://github.com/lxc/lxd/blob/master/doc/lxd-ssl-authentication.md might be helpful? Or am I missing a document covering the above?

- https://stgraber.org/2016/03/15/lxd-2-0-installing-and-configuring-lxd-212/
- https://github.com/lxc/lxd/issues/1858
- https://github.com/lxc/lxd/blob/master/doc/lxd-ssl-authentication.md

### PR DESCRIPTION
**Pull request by albertodonato**
Fixes #3448 

This renames the documentation file as suggested and extends it with answers from comments.

### CODE REVIEW

File: doc/security.md [hunk 1]
```diff
@@ -112,3 +122,15 @@ trusted.
 
 This happens if another trusted client or the local server administrator
 removed the trust entry on the server.
+
+
+# Production setup
+For production setup, it's reccomended that `core.trust_password` is unset
```
Comment (brauner) [inline doc/security.md]: s/reccomended/recommended/

File: doc/security.md [hunk 2]
```diff
@@ -112,3 +122,15 @@ trusted.
...
+after all clients have been added.
+
+This prevents brute-force attacks trying to guess the password.
```
Comment (brauner) [inline doc/security.md]: I guess this is meant as an addendum to the above sentence about unsetting trust password? :) If so, then remove the newline and make it one paragraph. :)

File: doc/security.md [hunk 3]
```diff
@@ -112,3 +122,15 @@ trusted.
...
+
+Also, `core.https_address` should be set to the single address where the server
```
Comment (brauner) [inline doc/security.md]: s/Also/Furthermore/

File: doc/security.md [hunk 4]
```diff
@@ -19,10 +19,13 @@ they're launched. The server will use that for all https connections to
 the LXD socket and the client will use its certificate as a client
 certificate for any client-server communication.
 
+To cause certificates to be regenerated, simply remove the old ones, new ones
```
Comment (brauner) [inline doc/security.md]: "To cause certificates to be regenerated, simply remove the old ones. On the next connection a new certificate will be generated."

**Comment by albertodonato** [comment]
@brauner thanks for pedantry :) All comments should be addressed

**Pull request merged**